router/core/context.go

```diff
@@ -253,11 +253,43 @@ type requestContext struct {
 	expressionContext expr.Context
 }
 
+func (c *requestContext) setErrorContext() {
+	if c.error == nil {
+		return
+	}
+	c.expressionContext.Request.Error = &ExprWrapError{c.error}
+}
+
+func (c *requestContext) ResolveAnyExpressionWithErrorOverride(expression *vm.Program, err error) (any, error) {
+	currContextRequest := c.expressionContext.Request
+	if err != nil {
+		// We don't really want to do a deep clone since we only know that error only can get modified
+		copyContext := expr.Context{
+			Request: expr.Request{
+				Auth:   currContextRequest.Auth,
+				URL:    currContextRequest.URL,
+				Header: currContextRequest.Header,
+				Error:  &ExprWrapError{err},
+			},
+		}
+		return expr.ResolveAnyExpression(expression, copyContext)
+	}
+
+	return c.ResolveAnyExpression(expression)
+}
+
+func (c *requestContext) ResolveAnyExpression(expression *vm.Program) (any, error) {
+	c.setErrorContext()
```

Comment:
Rather than doing it in every method, should we introduce a setter on the request context and do it there? I think you misunderstood my previous feedback.